test/buffer.jl

```diff
@@ -0,0 +1,70 @@
+using Serialization: deserialize, serialize
+using Test
+using ray_core_worker_julia_jll: LocalMemoryBuffer, Data, Size, OwnsData, IsPlasmaBuffer
+
+@testset "LocalMemoryBuffer" begin
+    data = UInt16[1:3;]
+
+    @testset "non-copied object reference" begin
+        buffer = LocalMemoryBuffer(pointer_from_objref(data), sizeof(data), false)
+        @test Data(buffer[]) == pointer_from_objref(data)
+        @test Size(buffer[]) == sizeof(data)
+        @test !OwnsData(buffer[])
+        @test !IsPlasmaBuffer(buffer[])
+
+        buffer_ptr = Data(buffer[]).cpp_object
+        result = unsafe_pointer_to_objref(buffer_ptr)
+        @test typeof(result) == typeof(data)
+        @test result === data
+    end
+
+    # TODO: Using `sizeof` is probably wrong for `pointer_from_objref` as there is probably
+    # additional Julia type metadata not being accounted for. This may be why we see a
+    # segfault when trying to use `pointer_from_objref`.
+    @testset "copied object reference" begin
+        buffer = LocalMemoryBuffer(pointer_from_objref(data), sizeof(data), true)
+        @test Data(buffer[]) != pointer_from_objref(data)
+        @test Size(buffer[]) == sizeof(data)
+        @test OwnsData(buffer[])
+        @test !IsPlasmaBuffer(buffer[])
+
+        # Attempting to use:
+        # `buffer_ptr = Data(buffer[]).cpp_object; unsafe_pointer_to_objref(buffer_ptr)`
+        # would result in a segfault.
+    end
```

Comment:
I'm vaguely concerned about this...seems like at some point we'll need to be able to construct these buffers from julia objects right?  or should we be serializing them _first_ and then creating the buffer from the array?